va/validation-authority.go

```diff
@@ -96,17 +97,37 @@ func (va ValidationAuthorityImpl) validateSimpleHTTP(identifier core.AcmeIdentif
 	}
 	hostName := identifier.Value
 
+	addrs, _, _, err := va.DNSResolver.LookupHost(hostName)
+	if err != nil {
+		challenge.Status = core.StatusInvalid
+		setChallengeErrorFromDNSError(err, &challenge)
+		va.log.Debug(fmt.Sprintf("%s [%s] DNS failure: %s", challenge.Type, identifier, err))
+		return challenge, challenge.Error
+	}
+	if len(addrs) == 0 {
+		challenge.Error = &core.ProblemDetails{
+			Type:   core.UnknownHostProblem,
+			Detail: fmt.Sprintf("Could not resolve %s", hostName),
+		}
+		challenge.Status = core.StatusInvalid
+		return challenge, errors.New(challenge.Error.Detail)
```

Comment:
Whoops, I'm going to blame this one on coding when you should be asleep...

On Wed, Jul 22, 2015 at 4:22 PM Jeff Hodges <notifications@github.com>
wrote:

> In va/validation-authority.go
> <https://github.com/letsencrypt/boulder/pull/505#discussion_r35276390>:
>
> > @@ -96,17 +97,37 @@ func (va ValidationAuthorityImpl) validateSimpleHTTP(identifier core.AcmeIdentif
> >  	}
> >  	hostName := identifier.Value
> >
> > +	addrs, _, _, err := va.DNSResolver.LookupHost(hostName)
> > +	if err != nil {
> > +		challenge.Status = core.StatusInvalid
> > +		setChallengeErrorFromDNSError(err, &challenge)
> > +		va.log.Debug(fmt.Sprintf("%s [%s] DNS failure: %s", challenge.Type, identifier, err))
> > +		return challenge, challenge.Error
> > +	}
> > +	if len(addrs) == 0 {
> > +		challenge.Error = &core.ProblemDetails{
> > +			Type:   core.UnknownHostProblem,
> > +			Detail: fmt.Sprintf("Could not resolve %s", hostName),
> > +		}
> > +		challenge.Status = core.StatusInvalid
> > +		return challenge, errors.New(challenge.Error.Detail)
>
> Why is challenge.Error not good enough here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/letsencrypt/boulder/pull/505/files#r35276390>.
>